Enable use of auditing with non-integer primary keys

rjw57 opened the pull request:
The audit models assume that the primary keys of the models being audited will be integers. This is not in general true. Move to storing the string representation of the primary key instead of the integer value.

Add a migration which alters the table. Postgres should be smart enough to coerce the values.

Required to allow auditing the IAR as described in uisautomation/iar-backend#36.

rjw57 committed:
Enable use of auditing with non-integer primary keys

The audit models assume that the primary keys of the models being
audited will be integers. This is not in general true. Move to storing
the string representation of the primary key instead of the integer
value.

Add a migration which alters the table. Postgres should be smart enough
to coerce the values.

Required to allow auditing the IAR.

rjw57 committed:
setup.py: bump version number

msb reviewed (changes requested):
as discuss - either bump or remove version

rjw57 committed:
middleware: improve compatibility with Django versions

We try to make our middleware compatible with old- and new-style
middleware. When used as old-style, the __init__ method is not passed
the "get_response" callable.

Work around this by defaulting get_response to None and asserting in the
new-style code paths that we were given an non-None get_response.
